src/sentry/integrations/discord/webhooks/message_component.py

```diff
@@ -0,0 +1,231 @@
+from __future__ import annotations
+
+from collections.abc import Mapping
+
+from rest_framework.response import Response
+
+from sentry.api.helpers.group_index.update import update_groups
+from sentry.integrations.discord.message_builder.base.base import DiscordMessageBuilder
+from sentry.integrations.discord.message_builder.base.component import (
+    DiscordComponentCustomIds as CustomIds,
+)
+from sentry.integrations.discord.message_builder.base.component.action_row import DiscordActionRow
+from sentry.integrations.discord.message_builder.base.component.select_menu import (
+    DiscordSelectMenu,
+    DiscordSelectMenuOption,
+)
+from sentry.integrations.discord.message_builder.base.flags import DiscordMessageFlags
+from sentry.integrations.discord.requests.base import DiscordRequest
+from sentry.integrations.discord.webhooks.handler import DiscordInteractionHandler
+from sentry.models.activity import ActivityIntegration
+from sentry.models.group import Group
+from sentry.services.hybrid_cloud.user.model import RpcUser
+
+from ..utils import logger
+
+NO_IDENTITY = "Sorry! You need to link your Discord account to your Sentry account to do that. You can do this with `/link`!"
+NOT_IN_ORG = "Sorry! You must be a member of the org this issue belongs to in order to act on it."
+ASSIGNEE_UPDATED = "Assignee has been updated."
+RESOLVE_DIALOG_OPTIONS = [
+    DiscordSelectMenuOption("Immediately", ""),
+    DiscordSelectMenuOption("In the next release", "inNextRelease"),
+    DiscordSelectMenuOption("In the current release", "inCurrentRelease"),
+]
+RESOLVED = "The issue has been resolved."
+RESOLVED_IN_NEXT_RELEASE = "The issue will be resolved in the next release."
+RESOLVED_IN_CURRENT_RELEASE = "The issue will be resolved in the current release."
+UNRESOLVED = "The issue has been unresolved."
+MARKED_ONGOING = "The issue has been marked as Ongoing."
+IGNORE_UNTIL_ESCALATES = "The issue will be ignored until it escalates."
+
+
```

Comment:
This is the new logic for handling interactions on notifications (Resolve button for example)